### ISSUE TITLE
Unsupported command argument type: TrueClass

### ISSUE THREAD

**Issue by 9mm** [rwojsznis/sidekiq-lock#21]
I feel like im going crazy, what is wrong with this code?

```
[1] pry(#<CampaignBatchTestJob>)> lock.acquire!
TypeError: Unsupported command argument type: TrueClass
```

```
class CampaignBatchTestJob

  include Sidekiq::Job
  include Sidekiq::Lock::Worker

  sidekiq_options queue: "default", retry: 0, lock: {
    name: proc {|id| "campaign_batch_test_job_#{id}"},
    timeout: 10.minutes.to_i * 1000, # millseconds
  }

  def perform(id, duping = false)
    binding.pry
    return unless lock.acquire!

```

```
    sidekiq (7.2.0)
      concurrent-ruby (< 2)
      connection_pool (>= 2.3.0)
      rack (>= 2.2.4)
      redis-client (>= 0.14.0)
    sidekiq-lock (0.6.0)
      redis (>= 3.0.5)
      sidekiq (>= 2.14.0)
```

**Comment by 9mm** [comment]
Exact stack trace:

```
TypeError: Unsupported command argument type: TrueClass
from /Users/zesty/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/redis-client-0.19.0/lib/redis_client/command_builder.rb:37:in `block in generate'
```

This other issue points to other issues with sidekiq 7 it seems

https://github.com/sidekiq-cron/sidekiq-cron/issues/378

But looks like they fixed it like this?

https://github.com/sidekiq-cron/sidekiq-cron/issues/378

**Comment by 9mm** [comment]
Yes, reverted to 7.1 and problem went away, it explains how to fix here:

https://github.com/sidekiq/sidekiq/blob/main/Changes.md#720

**Comment by rwojsznis** [comment]
Hey Travis! Thank you for your report, will try to look into it and release a new backward-compatible version _or_ a new breaking version that support 7.2 (as this gem didn't change for years and it will probably never will I guess that would be acceptable _as a last resort_)

**Comment by rwojsznis** [comment]
[Released 0.7.0](https://github.com/rwojsznis/sidekiq-lock/releases/tag/v0.7.0) which _should_ fix your problem

[please do give it a go](https://rubygems.org/gems/sidekiq-lock); I'm not actively using this gem anymore, but hopefully it still works as expected 😅
